src/Middleware/Spa/SpaProxy/src/SpaProxyMiddleware.cs

```diff
@@ -0,0 +1,92 @@
+    {
+        private readonly RequestDelegate _next;
+        private readonly SpaProxyLaunchManager _spaProxyLaunchManager;
+        private readonly IOptions<SpaDevelopmentServerOptions> _options;
+        private readonly IHostApplicationLifetime _hostLifetime;
+        private readonly ILogger<SpaProxyMiddleware> _logger;
+
+        public SpaProxyMiddleware(
+            RequestDelegate next,
+            SpaProxyLaunchManager spaProxyLaunchManager,
+            IOptions<SpaDevelopmentServerOptions> options,
+            IHostApplicationLifetime hostLifetime,
+            ILogger<SpaProxyMiddleware> logger)
+        {
+            _next = next ?? throw new ArgumentNullException(nameof(next));
+            _spaProxyLaunchManager = spaProxyLaunchManager ?? throw new ArgumentNullException(nameof(spaProxyLaunchManager));
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+            _hostLifetime = hostLifetime ?? throw new ArgumentNullException(nameof(hostLifetime));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            if (context.Request.Path.Equals(new Uri(_options.Value.ServerUrl).LocalPath))
+            {
+                context.Response.Headers[HeaderNames.CacheControl] = "no-cache, no-store, must-revalidate, max-age=0";
+                if (!await _spaProxyLaunchManager.IsSpaProxyRunning(context.RequestAborted))
+                {
+                    _spaProxyLaunchManager.StartInBackground(_hostLifetime.ApplicationStopping);
+                    _logger.LogInformation("SPA proxy is not ready. Returning temporary landing page.");
+                    context.Response.ContentType = "text/html";
+
+                    await using var writer = new StreamWriter(context.Response.Body, Encoding.UTF8);
+                    await writer.WriteAsync(GenerateSpaLaunchPage(_options.Value));
+                }
+                else
+                {
+                    _logger.LogInformation($"SPA proxy is ready. Redirecting to {_options.Value.ServerUrl}.");
+                    context.Response.Redirect(_options.Value.ServerUrl);
+                }
+            }
+            else
+            {
+                await _next(context);
+            }
+
+            string GenerateSpaLaunchPage(SpaDevelopmentServerOptions options)
+            {
+                return $@"
+<!DOCTYPE html>
+<html lang=""en"">
+<head>
+  <meta charset = ""UTF-8"" >
+  <meta http-equiv=""X-UA-Compatible"" content=""IE=edge"">
+  <meta http-equiv=""refresh"" content=""3"">
+  <meta name=""viewport"" content=""width=device-width, initial-scale=1.0"">
+  <title>SPA proxy launch page</title>
+</head>
+<body>
+  <h1>Launching the SPA proxy...</h1>
+  <p>This page will automatically redirect to <a href=""{options.ServerUrl}"">{options.ServerUrl}</a> when the SPA proxy is ready.</p>
```

Comment:
escaping?

The text implies the navigation is automatic, why give a link? It's probably better to remove the href and only show the url text.

Reply:
The navigation is automatic (the client polls with the `http-equiv` meta tag every three seconds). You can click the link to go to the SPA front-end directly if you don't want to wait. We display this page to keep VS happy while launching the SPA CLI proxy.

Reply:
Then you'll at least need to properly escape the value.

Reply:
@Tratcher I already did, are you referring to something else?

BTW, this will just hold `https://localhost:5002`